sysdig/resource_sysdig_secure_cloud_auth_account.go

```diff
@@ -387,26 +387,42 @@ func constructAccountComponents(data *schema.ResourceData) []*cloudauth.AccountC
 					if data.Get(SchemaCloudProviderType).(string) == cloudauth.Provider_PROVIDER_GCP.String() {
 						spGcp := &internalServicePrincipalMetadata{}
 						err = json.Unmarshal([]byte(value.(string)), spGcp)
-						if len(spGcp.Gcp.Key) >= 0 {
+
+						hasKeyData := len(spGcp.Gcp.Key) > 0
+						hasWifData := spGcp.Gcp.WorkloadIdentityFederation != nil && len(spGcp.Gcp.Email) > 0
+
+						if !hasKeyData && !hasWifData {
+							diag.FromErr(fmt.Errorf("failed to parse one of either service principal types, 'key' or 'workload_identity_federation'+'email'"))
+							continue
+						}
+
+						metadata := &cloudauth.AccountComponent_ServicePrincipalMetadata{
+							ServicePrincipalMetadata: &cloudauth.ServicePrincipalMetadata{
+								Provider: &cloudauth.ServicePrincipalMetadata_Gcp{
+									Gcp: &cloudauth.ServicePrincipalMetadata_GCP{},
+								},
+							},
+						}
+
+						if hasKeyData {
 							var spGcpKeyBytes []byte
 							spGcpKeyBytes, err = base64.StdEncoding.DecodeString(spGcp.Gcp.Key)
 							if err != nil {
 								diag.FromErr(err)
 							}
 							spGcpKey := &cloudauth.ServicePrincipalMetadata_GCP_Key{}
 							err = json.Unmarshal(spGcpKeyBytes, spGcpKey)
-							component.Metadata = &cloudauth.AccountComponent_ServicePrincipalMetadata{
-								ServicePrincipalMetadata: &cloudauth.ServicePrincipalMetadata{
-									Provider: &cloudauth.ServicePrincipalMetadata_Gcp{
-										Gcp: &cloudauth.ServicePrincipalMetadata_GCP{
-											Key:                        spGcpKey,
-											WorkloadIdentityFederation: spGcp.Gcp.WorkloadIdentityFederation,
-											Email:                      spGcp.Gcp.Email,
-										},
-									},
-								},
+							metadata.ServicePrincipalMetadata.GetGcp().Key = spGcpKey
+						}
+
+						if hasWifData {
+							metadata.ServicePrincipalMetadata.GetGcp().WorkloadIdentityFederation = &cloudauth.ServicePrincipalMetadata_GCP_WorkloadIdentityFederation{
+								PoolProviderId: spGcp.Gcp.WorkloadIdentityFederation.PoolProviderId,
```

Comment:
Not a blocker, but since we are updating this code, can we ensure passing all the fields within `GCP_WorkloadIdentityFederation` as received, rather than passing just one of them i.e `PoolProviderId` ?

Reply:
This comment is a noop now, with the change block removed.